### PR TITLE
cleanup zen instances before conversion

### PR DESCRIPTION
**Pull request by bluzarraga**
Signed-off-by: Ben Luzarraga <luzarragaben@ibm.com>

Cleanup zen instances before conversion process by patching out the csNamespace key in zenservice and deleting the iam config job and the zen client in cloud pak namespace. 

Open to feedback, related to https://github.com/IBM/ibm-common-service-operator/pull/905 and https://github.com/IBM/ibm-common-service-operator/pull/909. Will likely be merge conflicts

Sample output:
```
zenservice.zen.cpd.ibm.com/iaf-zen-cpdservice patched
job.batch "iam-config-job" deleted
client.oidc.security.ibm.com "zenclient-cpns1" deleted
zenservice.zen.cpd.ibm.com/iaf-zen-cpdservice patched
[INFO] CS Namespace not defined in iaf-zen-cpdservice in cpns2. Moving on...
No zen client in cpns2. Moving on...
```

### CODE REVIEW

File: convert_to_multi_instance.sh [hunk 1]
```diff
@@ -84,6 +84,8 @@ function prepare_cluster() {
     # TODO for more advanced checking
     # find all namespaces with cs-operator running
     # each namespace should be in configmap
+    
+    cleanupZenService $cm_name
```
Comment (bitscuit) [inline convert_to_multi_instance.sh:88]: move this to after the NSS CRs are removed according to discussion with Simon

**Comment by bluzarraga** [comment]
Looking at error handling nss. As it is now, if the nss specified in the removeNSS function are not present, the script breaks. If it makes sense to include it in this pr I will otherwise I will open another

**Comment by bitscuit** [comment]
fix in a new PR for better tracking

**Comment by bluzarraga** [comment]
nss error handling https://github.com/IBM/ibm-common-service-operator/pull/917

**Comment by bitscuit** [comment]
I think we can change the function to use yq, similar to `check_cm_ns_exist` function, to read the namespaces

**Review by bitscuit — APPROVED** [review]
/lgtm

**Comment by bluzarraga** [comment]
Need to add some kind of error handling/check for zen resources on a system. If there aren't any zen resources the script will fail trying to clean them up

**Comment by bluzarraga** [comment]
Yes, if a zenclient or zenservice does not exist in a namespace the return is `No resources found in ibm-common-services namespace.` but it does not trigger the `echo "failed"` so the `return_value` is `""` instead of `failed`. We would either need to grep on something and it return nothing (which could involve a little guess work) or look for an empty response.

**Pull request merged**